index.d.ts

```diff
@@ -46,7 +56,7 @@ export interface Action<T = any> {
  * @template S The type of state consumed and produced by this reducer.
  * @template A The type of actions the reducer can potentially respond to.
  */
-export type Reducer<S = any, A extends Action = Action> = (state: S | undefined, action: A) => S;
+export type Reducer<S = any, A extends Action = AnyAction> = (state: S | undefined, action: A) => S;
```

Comment:
Actually, this is not needed, because any interface with additional properties extends `Action` interface.

Reply:
The default type for `A` is needed so that any reducer acting on the state `State` would be assignable to `Reducer<State>` (with omitted action type parameter).

You may want to change it back to `Action` and check out the tests that break.

Reply:
Ah, I see now. Thanks.